metagamer/environments/tictactoe.py

```diff
@@ -45,25 +51,26 @@ def check_win(board: np.array) -> int:
         board (np.array): 3 x 3 board
 
     Returns:
-        int: 0  -> no winner
-             1  -> player 1
+        int:  0 -> no winner
+              1 -> player 1
              -1 -> player 2
 
     """
-    assert board.shape == (3, 3)
+    assert board.shape[0] == board.shape[1]
+    compval = board.shape[0]
     for axis in [0, 1]:
         axis_sum = board.sum(axis=axis)
-        if 3 in axis_sum:
+        if compval in axis_sum:
             return 1
-        elif -3 in axis_sum:
+        elif -compval in axis_sum:
             return -1
 
     diag = board.diagonal().sum()
     diag_inv = np.fliplr(board).diagonal().sum()
     for diag_sum in [diag, diag_inv]:
-        if diag_sum == 3:
+        if diag_sum == compval:
             return 1
-        elif diag_sum == -3:
+        elif diag_sum == -compval:
```

Comment:
This all looks like sensible generalising and tidying

Reply:
This was meant to help support the n,n boards being proposed a while back

Reply:
Which I think is still a very good (and potentially very easy) expansion of the game.